algorithms/src/polycommit/sonic_pc/mod.rs

```diff
@@ -266,22 +266,25 @@ impl<E: PairingEngine, S: AlgebraicSponge<E::Fq, 2>> SonicKZG10<E, S> {
     pub fn combine_for_open<'a>(
         universal_prover: &UniversalProver<E>,
         ck: &CommitterUnionKey<E>,
-        labeled_polynomials: impl IntoIterator<Item = &'a LabeledPolynomial<E::Fr>>,
+        labeled_polynomials: impl ExactSizeIterator<Item = &'a LabeledPolynomial<E::Fr>>,
         rands: impl IntoIterator<Item = &'a Randomness<E>>,
         fs_rng: &mut S,
     ) -> Result<(DensePolynomial<E::Fr>, Randomness<E>), PCError>
     where
         Randomness<E>: 'a,
         Commitment<E>: 'a,
     {
-        Ok(Self::combine_polynomials(labeled_polynomials.into_iter().zip_eq(rands).map(|(p, r)| {
+        let mut to_combine = Vec::with_capacity(labeled_polynomials.len());
+
+        for (p, r) in labeled_polynomials.zip_eq(rands) {
```

Comment:
This `.zip_eq` needs an `ensure!` on the `len()`.

Reply:
the rebase took care of that ([L278](https://github.com/AleoHQ/snarkVM/pull/2247/files#diff-1c2269e15b2c9eb1b102866e39b60f7149b01d43503a3bdd02081fc7200a358cR278))